### ISSUE TITLE
What is the desired behaviour when input_spec isn't provided to Workflow __init__?

### ISSUE THREAD

**Issue by tclose** [nipype/pydra#491]
https://github.com/nipype/pydra/blob/e0ea60d4c55c1f25951c2d931325159009d8d058/pydra/engine/core.py#L796-L817

If an `input_spec` isn't provided to Workflow.__init__ what should happen, an empty `input_spec` generated? Because as it stands `self.input_spec` isn't set and the method gets stuck in an infinite loop trying to access the attribute that isn't there.

**Comment by djarecka** [comment]
@tclose - thanks for reporting this! we should raise an exception. Would you be interested in creating PR?

**Comment by tclose** [comment]
Sure, do you want an empty `input_spec` created in the case when it isn't provided?

**Comment by djarecka** [comment]
No, I think we want to have an error, we do want to have at least one input for every single workflow and discourage people from hard-coding input paths etc. Unless, you have an example that you can show it will be useful

**Comment by tclose** [comment]
#497 addresses this issue

**Comment by tclose** [comment]
One thing that seems a little counterintuitive to me unless I don't understand how it works (quite likely) is the ability to define both the input_spec and input values in the Workflow init.

I was expecting a Workflow to conceptually map onto a task (i.e. an "uninstantiated node", sorry not sure what nomeclature you use), which then could be parameterised for different applications. However, it seems that you are parameterising it as it is defined, do I have that right?

**Comment by djarecka** [comment]
I think the main reason for `input_spec` was that we allow for input provided as kwarg to `Workflow`, but perhaps we might rething
